### ISSUE TITLE
Add Color List for the "Toggle Highlight of All Occurrences"

### ISSUE THREAD

**Issue by alvolturbo** [krasa/GrepConsole#185]
**Add Color List**
Is is it possible to add preferred Colors List to make Toggle Highlight according to it?
If there is not enough item in the list, plugin could be used with random colors as it is for now.
Removing any color highlight could add item to the preferred list again.

**Comment by krasa** [comment]
Try https://plugins.jetbrains.com/plugin/9511-multihighlight - it did not support plaintext when I made this feature.

**Comment by alvolturbo** [comment]
thanks
i know this plugin but it highlights only full word, not exactly that pattern i have selected (but it has cyclic color switch what is good)

**Comment by alvolturbo** [comment]
is it possible to make live highlights for all instances of that pattern i have selected? 
changing selection--changing other places with that highlighting

**Comment by krasa** [comment]
https://plugins.jetbrains.com/plugin/201-browsewordatcaret does that but only for whole words, you could probabaly change that

**Comment by alvolturbo** [comment]
this is not what i mean... 
any pattern with several words and any characters in that pattern very useful feature to compare code parts "on fly" (when highlight becomes not active)
but thanks in any case

**Comment by krasa** [comment]
Sorry, I have no idea what you mean.

**Comment by alvolturbo** [comment]
like this
https://user-images.githubusercontent.com/9571591/107694395-fe86e380-6cb7-11eb-9f42-6587dc914628.mp4

**Comment by krasa** [comment]
yes, that's what I meant, this could do it https://github.com/minman/browse-word-at-caret/compare/master...krasa:custom?expand=1
![q](https://user-images.githubusercontent.com/1160875/107712013-ebc8da80-6cc8-11eb-9cda-efd35dfd04e4.gif)

**Comment by alvolturbo** [comment]
great! good feature
can you advise of how to install your "custom brach update" to the current plugin "browse-word-at-caret"?

**Comment by krasa** [comment]
https://github.com/krasa/browse-word-at-caret/releases/download/3.0.5-custom/browse-word-at-caret.jar
You might want to get into plugin development. I commited some project files, so the the process should be similar to this: https://github.com/krasa/StringManipulation#how-to-build-and-run

**Comment by alvolturbo** [comment]
thanks a lot for you tutor
